src/Nethermind/Nethermind.Optimism/CL/Driver.cs

```diff
@@ -47,49 +51,65 @@ public Driver(IL1Bridge l1Bridge,
                 {
-                    if (_logger.IsWarn)
-                        _logger.Warn(
-                            $"Derived batch is out of order. Highest derived block: {_currentDerivedBlock}, Batch first block: {firstBlockNumber}");
-                    throw new ArgumentException("Batch is out of order");
+                    (BatchV1 decodedBatch, ulong batchOrigin) = await _decodingPipeline.DecodedBatchesReader.ReadAsync(token);
+                    await ProcessDecodedBatch(decodedBatch, batchOrigin, token);
+                    continue;
                 }
 
-                L2Block l2Parent = await _l2Api.GetBlockByNumber(firstBlockNumber - 1);
-
-                var derivedPayloadAttributes = _derivationPipeline
-                    .DerivePayloadAttributes(l2Parent, decodedBatch, token)
-                    .GetAsyncEnumerator(token);
-                while (await derivedPayloadAttributes.MoveNextAsync())
+                if (l1BridgeStep.IsCompleted)
                 {
-                    PayloadAttributesRef payloadAttributes = derivedPayloadAttributes.Current;
-                    bool valid = await _executionEngineManager.ProcessNewDerivedPayloadAttributes(payloadAttributes);
-                    if (!valid)
+                    L1BridgeStepResult result = l1BridgeStep.Result;
+                    _logger.Warn(result.ToString());
+                    switch (result.Type)
                     {
-                        if (_logger.IsWarn) _logger.Warn($"Derived invalid Payload Attributes. {payloadAttributes}");
-                        break;
+                        case L1BridgeStepResultType.Block:
+                        {
+                            foreach (DaDataSource daDataSource in result.NewData!)
+                            {
+                                await _decodingPipeline.DaDataWriter.WriteAsync(daDataSource, token);
+                            }
+                            l1BridgeStep = _l1Bridge.Step(token);
+                            break;
+                        }
+                        case L1BridgeStepResultType.Finalization:
+                        {
+                            await ProcessNewFinalized(result.NewFinalized!.Value, token);
+                            l1BridgeStep = _l1Bridge.Step(token);
+                            break;
+                        }
+                        case L1BridgeStepResultType.Reorg:
+                        {
+                            await ProcessReorg(token);
+                            l1BridgeStep = _l1Bridge.Step(token);
+                            break;
+                        }
+                        case L1BridgeStepResultType.Skip:
+                        {
+                            l1BridgeStep = Task.Run(async () =>
+                            {
+                                await Task.Delay(120000, token);
+                                return await _l1Bridge.Step(token);
+                            });
```

Comment:
Could we keep the `Task.Delay` inside the `case` statement, and extract the common `l1BridgeStep = _l1BridgeStep.Step(token)` outside the `switch` block?